esrally/client/factory.py

```diff
@@ -28,13 +28,23 @@
 
 class EsClientFactory:
     """
-    Abstracts how the Elasticsearch client is created. Intended for testing.
+    Abstracts how the Elasticsearch client is created and customizes the client for backwards
+    compatibility guarantees that are broader than the library's defaults.
     """
 
-    def __init__(self, hosts, client_options):
-        self.hosts = hosts
+    def __init__(self, hosts, client_options, distribution_version=None):
+        # We need to pass a list of connection strings to the client as of elasticsearch-py 8.0
```

Comment:
We mention the new version of elasticsearch-py but the names are all different. Sometimes there are quotes, sometimes we talk about 8.0, sometimes 8.x, sometimes the version number comes before the library name, etc.

Can we please:

 * Only mention that library name when there's a if/else or try/except? Here I don't think the comment is needed. This is the way to pass connection strings, the fact that it's new will be irrelevant when this pull request will be merged.
 * When mentioning the name, can we use a single way to refer to it? It will make removing all those if/else and try/except much easier in the future.

Reply:
++,  I've simplified on `'elasticsearch-py' 8.x`. 

See https://github.com/elastic/rally/pull/1669/commits/2d3270ca898e0d0101e4020535c84b86941f5207